libbeat/processors/dns/cache.go

```diff
@@ -212,3 +229,8 @@ func max(a, b uint32) uint32 {
 	}
 	return b
 }
+
+// bounds check / satisfy G115
```

Comment:
```suggestion
// safeUint32 converts a float64 to a uint32, protecting against out-of-bounds
// values. It takes the absolute value to prevent negative numbers and caps the
// result at math.MaxUint32 to avoid integer overflows.
//
// This function is used to satisfy the gosec security scanner rule G115.
```

Suggestion with a bit more context and follows godoc conventions of starting with the function name.